File: bug61_redundant_assignment/bug61_redundant_assignment/main.java
public class main {
    public static void main(String[] args) {
        long acc = 0x9191919191919191L;
        for (int i = 0; i < 10; i++) {
            acc ^= i * 0x9E3779B97F4A7C15L;
            acc = ((acc << (i % 13 + 1)) | (acc >>> (63 - (i % 13))));
            acc = acc;
            acc = (acc * 0xBF58476D1CE4E5B9L);
            acc = 0x9191919191919191L;
        }
        
        if (acc == 0x9191919191919191L) {
            System.err.println("ValueError: redundant final assignment reset acc to seed each iteration");
            System.exit(1);
        }
    }
}

File: bug61_redundant_assignment/bug61_redundant_assignment/main.js
function main() {
    let acc = 0x9191919191919191n;
    for (let i = 0n; i < 10n; i++) {
        acc ^= i * 0x9E3779B97F4A7C15n;
        acc &= 0xFFFFFFFFFFFFFFFFn;
        acc = ((acc << (i % 13n + 1n)) | (acc >> (63n - i % 13n))) & 0xFFFFFFFFFFFFFFFFn;
        acc = acc;
        acc = (acc * 0xBF58476D1CE4E5B9n) & 0xFFFFFFFFFFFFFFFFn;
        acc = 0x9191919191919191n;
    }
    
    if (acc === 0x9191919191919191n) {
        console.error("ValueError: redundant final assignment reset acc to seed each iteration");
        process.exit(1);
    }
}
main();
